bitcoin/short_channel_id.c

```diff
@@ -48,7 +49,7 @@ bool short_channel_id_from_str(const char *str, size_t strlen,
 
 #ifdef COMPAT_V062
 	/* Pre-adelaide format vs. post-adelaide format */
-	if (strchr(buf, ':'))
+	if (may_be_deprecated_form && strchr(buf, ':'))
```

Comment:
My only thought: It *might* be cleaner if we communicated `deprecated_apis` (which is a global) to the subdaemons and then use that  here directly instead of `may_be_deprecated`. But it might not.